### DISCUSSION TITLE
How to obtain default texture of material for later reference

### DISCUSSION THREAD

**Discussion by henrik-dons** [google/model-viewer#4936] [Q&A]
Hello all,

I want to collect the default texture and store for later reference for the reset button functionality

Idea is to dynamically allow to change textures, and use reset button to change back to default model texture

How can I collect the default material texture?

Looking forward to a reply

Cheers

**Comment by danilokrulj** [comment]
Why don't you just save the default textures in the same folder? I have the default textures saved like the rest that are being applied to the model. Just open the model in a 3D software and export the textures from there, after that use them like any other textures. Maybe there is a better way to do this but until someone smarter shows up this is what I suggest you to do. :)

**Comment by elalish** [comment]
You can follow this [example](https://modelviewer.dev/examples/scenegraph/#createTexturesExample) - the only difference is to save e.g. `const default = material[channel].texture;` somewhere and then use it again sometime e.g. `material[channel].setTexture(default);`

**Comment by henrik-dons** [comment]
Thank you guys! I will give that a try
